source/templates/stack.slim

```diff
@@ -6,10 +6,10 @@
 		article
 			section
 				.max-width-wrapper
-					p.notification.warning.stack-error[ng-if="Stack.all.indexOf(appService.stack) == -1"] == data[:strings][:stack][:invalid_stack]
+					p.notification.warning.stack-error[ng-if="stackCtrl.canShowDefaultInvalidSelectorError()"] == data[:strings][:stack][:invalid_stack]
 					p.notification.warning.stack-error[ng-if="Stack.all.indexOf(appService.stack) != -1 && appService.availableStacks().indexOf(appService.stack) == -1"] == data[:strings][:stack][:incompatible_stack]
 					.stack-selector-wrapper
-						select.stack-selector[ng-model="appService.stack" ng-options="stackCtrl.stackOptionTitle(stack) for stack in stackCtrl.stackOptions()" ng-change="stackCtrl.onDefaultStackChange()"]
+						select.stack-selector[ng-if="stackCtrl.canShowStackSelector()" ng-model="appService.stack" ng-options="stackCtrl.stackOptionTitle(stack) for stack in stackCtrl.stackOptions()" ng-change="stackCtrl.onDefaultStackChange()"]
```

Comment:
By the above suggestion, these changes will not be needed: the original check for inclusion in `Stack.all` will be sufficient; the `canShowStackSelector` was afaik making sure the agent pool stacks are loaded - having the agent pool stacks loaded in the `Stack.getAll()` guarantees that it'll be already loaded by the time this whole tab is displayed.